src/zcbor_common.c

```diff
@@ -130,3 +151,74 @@ bool zcbor_new_state(zcbor_state_t *state_array, uint_fast32_t n_states,
 	}
 	return true;
 }
+
+void zcbor_update_state(zcbor_state_t *state,
+		const uint8_t *payload, size_t payload_len)
+{
+	state->payload = payload;
+	state->payload_end = payload + payload_len;
+
+	update_backups(state, state->payload_end);
+}
+
+
+bool zcbor_validate_string_fragments(struct zcbor_string_fragment *fragments,
+		uint_fast32_t num_fragments)
+{
+	size_t total_len = 0;
+
+	if (!fragments) {
```

Comment:
The NULL check below does `== NULL` instead of `!fragments[i].fragment.value`, we should stick to one or the other. I suggest having explicit `== NULL` checks.